src/libraries/System.Text.Json/src/System/Text/Json/JsonHelpers.Escaping.cs

```diff
@@ -53,6 +53,34 @@ public static byte[] GetEscapedPropertyNameSection(ReadOnlySpan<byte> utf8Value,
             return escapedString;
         }
 
+        public static string GetPropertyName(
+            string rawPropertyName,
+            JavaScriptEncoder? encoder)
+        {
+            int indexOfFirstCharacterToEncode;
+            string returnValue;
+
+            indexOfFirstCharacterToEncode = JsonWriterHelper.NeedsEscaping(rawPropertyName.AsSpan(), encoder);
+
+            if (indexOfFirstCharacterToEncode != -1)
+            {
+                returnValue = GetEscapedPropertyName(rawPropertyName, encoder);
+            }
+            else
+            {
+                returnValue = rawPropertyName;
+            }
+
+            return returnValue;
+        }
+
+        private static string GetEscapedPropertyName(
+            string rawPropertyName,
+            JavaScriptEncoder? encoder)
+        {
+            return (encoder ?? JavaScriptEncoder.Default).Encode(rawPropertyName);
+        }
+
```

Comment:
It doesn't seem like the `encoder` parameter is ever null:
```suggestion
        public static string GetEscapedPropertyName(
            string rawPropertyName,
            JavaScriptEncoder encoder)
        {
            int indexOfFirstCharacterToEncode = JsonWriterHelper.NeedsEscaping(rawPropertyName.AsSpan(), encoder);
            return indexOfFirstCharacterToEncode >= 0
                ? encoder.Encoder(rawPropertyName)
                : rawPropertyName;
        }
```